File: app/src/main/java/com/example/artikproject/background_work/datebase/MySharedPreferences.java
package com.example.artikproject.background_work.datebase;

import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;

public class MySharedPreferences {

    public static void put(Context context, String name, int value){
        SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
        SharedPreferences.Editor edit = sharedPreferences.edit();
        edit.putInt(name, value);
        edit.apply(); //apply
    }

    public static void put(Context context, String name, String value){
        SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
        SharedPreferences.Editor edit = sharedPreferences.edit();
        edit.putString(name, value);
        edit.apply(); //apply
    }

    public static void put(Context context, String name, boolean value){
        SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
        SharedPreferences.Editor edit = sharedPreferences.edit();
        edit.putBoolean(name, value);
        edit.apply(); //apply
    }

    public static void remove(Context context, String name){
        SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
        SharedPreferences.Editor edit = sharedPreferences.edit();
        edit.remove(name);
        edit.apply(); //apply
    }

    public static String get(Context context, String name, String default_value){
        return PreferenceManager.getDefaultSharedPreferences(context)
                .getString(name, default_value);
    }

    public static int get(Context context, String name, int default_value){
        return PreferenceManager.getDefaultSharedPreferences(context)
                .getInt(name, default_value);
    }
}

File: app/src/main/java/ru/agpu/artikproject/background_work/datebase/MySharedPreferences.kt
package ru.agpu.artikproject.background_work.datebase

import android.content.Context
import android.preference.PreferenceManager

object MySharedPreferences {
    fun putPref(context: Context?, name: String?, value: Int) {
        val sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context)
        val edit = sharedPreferences.edit()
        edit.putInt(name, value)
        edit.apply() //apply
    }

    fun putPref(context: Context?, name: String?, value: String?) {
        val sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context)
        val edit = sharedPreferences.edit()
        edit.putString(name, value)
        edit.apply() //apply
    }

    fun putPref(context: Context?, name: String?, value: Boolean) {
        val sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context)
        val edit = sharedPreferences.edit()
        edit.putBoolean(name, value)
        edit.apply() //apply
    }

    fun removePref(context: Context?, name: String?) {
        val sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context)
        val edit = sharedPreferences.edit()
        edit.remove(name)
        edit.apply() //apply
    }

    fun getPref(context: Context?, name: String?, default_value: String): String {
        return PreferenceManager.getDefaultSharedPreferences(context).getString(name, default_value) ?: ""
    }

    fun getPref(context: Context?, name: String?, default_value: Int): Int {
        return PreferenceManager.getDefaultSharedPreferences(context).getInt(name, default_value)
    }

    fun getPref(context: Context?, name: String?, default_value: Boolean): Boolean {
        return PreferenceManager.getDefaultSharedPreferences(context).getBoolean(name, default_value)
    }
}